### PR TITLE
Improve phpstan to level 1

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because master will be released soon and it is easier.

### CODE REVIEW

File: src/Model/Gallery.php [hunk 1]
```diff
@@ -50,7 +50,8 @@ abstract class Gallery implements GalleryInterface
     protected $defaultFormat = MediaProviderInterface::FORMAT_REFERENCE;
 
     /**
-     * @var GalleryItemInterface[]|Collection
+     * @var Collection
+     * @phpstan-var Collection<GalleryItemInterface>
```
Comment (VincentLanglet) [inline src/Model/Gallery.php:54]:
```suggestion
     * @phpstan-var Collection<array-key, GalleryItemInterface>
```

And @phpstan-var is not really needed, phpstorm understand this. `Foo<bar>`

File: src/Model/GalleryInterface.php [hunk 2]
```diff
@@ -100,12 +102,14 @@ public function setDefaultFormat($defaultFormat);
     public function getDefaultFormat();
 
     /**
-     * @param array $galleryItems
+     * @param Collection $galleryItems
+     * @phpstan-param Collection<GalleryItemInterface> $galleryItems
```
Comment (VincentLanglet) [inline src/Model/GalleryInterface.php:106]: same

File: src/Model/GalleryInterface.php [hunk 3]
```diff
@@ -100,12 +102,14 @@ public function setDefaultFormat($defaultFormat);
...
      */
     public function setGalleryItems($galleryItems);
 
     /**
-     * @return GalleryItemInterface[]
+     * @return Collection
+     * @phpstan-return Collection<GalleryItemInterface>
```
Comment (VincentLanglet) [inline src/Model/GalleryInterface.php:112]: same

File: tests/Provider/BaseProviderTest.php [hunk 4]
```diff
@@ -16,19 +16,16 @@
...
-use Sonata\MediaBundle\Provider\MediaProviderInterface;
+use Sonata\MediaBundle\Tests\App\Provider\TestProvider;
 use Sonata\MediaBundle\Tests\Entity\Media;
 use Sonata\MediaBundle\Thumbnail\ThumbnailInterface;
-use Symfony\Component\Form\FormBuilderInterface;
 
 class BaseProviderTest extends AbstractProviderTest
 {
-    public function getProvider(): MediaProviderInterface
+    public function getProvider(): TestProvider
```
Comment (jordisala1991) [inline tests/Provider/BaseProviderTest.php:28]: How show I deal with this? this method is abstract and returns an interface, but the concrete classes have more methods and phpstan complains, but if I do this psalm complains on higher levels, and php 7.3 complains too.

Comment (VincentLanglet) [reply tests/Provider/BaseProviderTest.php:28]: Which method do you need more than the one in MediaProviderInterface ?

Quick solution would be to keep the `MediaProviderInterface` return type and add `@return TestProvider`, it solve the php 7.3 error. But psalm will complain on higher level I think.

Right solution would be to make the `AbstractProviderTest` using a template
`@phpstan-template P of MediaProviderInterface` and
```
/** @phpstan-return P */
public function getProvider(): MediaProviderInterface;
```

Comment (jordisala1991) [reply tests/Provider/BaseProviderTest.php:28]: I will go for the template solution, seems to be the most accurate

Comment (jordisala1991) [reply tests/Provider/BaseProviderTest.php]: Done

File: src/Model/GalleryInterface.php (hunk 2 above)
Comment (jordisala1991) [reply src/Model/GalleryInterface.php]: Fixed

File: src/Model/Gallery.php (hunk 1 above)
Comment (jordisala1991) [reply src/Model/Gallery.php]: Fixed

File: src/Model/GalleryInterface.php (hunk 3 above)
Comment (jordisala1991) [reply src/Model/GalleryInterface.php]: Fixed

**Review by VincentLanglet — APPROVED** [review]

**Review by phansys — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks

**Pull request merged**